x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/ConfigInitialNode.java

```diff
@@ -168,38 +173,31 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
             // this will show a message to the console (the first time the node starts) and auto-configuration is effectively bypassed
             // the message will not be subsequently shown (because auto-configuration doesn't run for node restarts)
-            if (options.has(strictOption)) {
-                throw new UserException(ExitCodes.CANT_CREATE, "Could not create auto configuration directory", e);
-            } else {
-                return; // silent error because we wish the node to start as usual (skip auto config) if config dir is not writable
-            }
+            throw new UserException(ExitCodes.CANT_CREATE, "Could not create auto configuration directory", e);
         }
 
-        // Ensure that the files created by the auto-config command MUST have the same owner as the config dir itself,
-        // as well as that the replaced files don't change ownership.
-        // This is because the files created by this command have hard-coded "no" permissions for "group" and "other"
-        UserPrincipal newFileOwner = Files.getOwner(instantAutoConfigDir, LinkOption.NOFOLLOW_LINKS);
-        if ((false == newFileOwner.equals(Files.getOwner(env.configFile(), LinkOption.NOFOLLOW_LINKS))) ||
-                (false == newFileOwner.equals(Files.getOwner(ymlPath, LinkOption.NOFOLLOW_LINKS))) ||
-                (Files.exists(keystorePath) && false == newFileOwner.equals(Files.getOwner(keystorePath, LinkOption.NOFOLLOW_LINKS)))) {
-            Files.deleteIfExists(instantAutoConfigDir);
-            if (options.has(strictOption)) {
-                throw new UserException(ExitCodes.CONFIG, "Aborting auto configuration because it would change config file owners");
-            } else {
-                return; // if a different user runs ES compared to the user that installed it, auto configuration will not run
-            }
+        // Check that the created auto-config dir has the same owner as the config dir.
+        // This is a sort of sanity check.
+        // If the node process works OK given the owner of the config dir, it should also tolerate the auto-created config dir,
+        // provided that they both have the same owner and permissions.
+        final UserPrincipal newFileOwner = Files.getOwner(instantAutoConfigDir, LinkOption.NOFOLLOW_LINKS);
+        if (false == newFileOwner.equals(Files.getOwner(env.configFile(), LinkOption.NOFOLLOW_LINKS))) {
+            deleteDirectory(instantAutoConfigDir);
+            // the following is only printed once, if the node starts successfully
+            throw new UserException(ExitCodes.CONFIG, "Aborting auto configuration because of config dir ownership mismatch");
         }
 
         // the transport key-pair is the same across the cluster and is trusted without hostname verification (it is self-signed),
         final X500Principal certificatePrincipal = new X500Principal(buildDnFromDomain(System.getenv("HOSTNAME")));
```

Comment:
The `HOSTNAME ` env var might not always be there in linux ( I think, I've never seen it happen ) but it won't be there in windows for sure. We could try to do something portable like `InetAddress.getLocalHost().getHostName()`  or check COMPUTERNAME in windows  but both could possibly fail and  -what's important- we don't _actually need_ this. It's nice for informational purposes when someone checks the certificate but subject is not used for hostname verification by browsers/clients

Reply:
It should be available, handled like your describing in the elasticsearch env script or it's ilk. It is relied upon in a couple of places already.

Reply:
Ha! I had totally missed that, thanks :)